### DISCUSSION TITLE
Artillery runs in 2 Jenkins Jobs

### DISCUSSION THREAD

**Discussion by vichybaby** [artilleryio/artillery#1237] [Artillery]
In my project, I need run performance test at 2 places
1. In Jenkins CI Pipeline - an informal short-time run in dev environment with low TPS
2. In Jenkins Job with Parameters Input Page - a formal long-time run in performance environment with high TPS

That means I cannot hardcode the 'environment', 'phases.duration', 'phases.arrivalRate' anywhere.

I am wondering if there is a solution: if Jenkins is passing values via Environment Variables, use them; otherwise, use default value

**Comment by hassy** [comment]
hi @vichybaby 👋 

There's a couple of ways you could do it.

1. The name of the `environment` itself can be a parameter, which can then be provided to the `artillery` command as an environment variable. For example, if the parameter name is `ARTILLERY_ENVIRONMENT` the command would then be:

    ```
    artillery run --environment $ARTILLERY_ENVIRONMENT my-test.yaml
    ```

2. The values of a phase can be set dynamically through environment variables too, and referenced in the Artillery script through `$processEnvironment` variable ([docs](https://www.artillery.io/docs/guides/guides/test-script-reference#environment-varables-in-config))

I put together a small example of what it could look like here: https://github.com/artilleryio/artillery-examples/tree/main/script-overrides
